### ISSUE TITLE
Change segment filename format

### ISSUE THREAD

**Issue by chrusher** [ekimekim/wubloader#7]
To potentially make everyone's lives easier, I suggest including the full datetime in the segment file names rather than just the minutes and second.

I can see a few advantages to this. First, it is easy to see if segments end up in the wrong hour directory. Second, if segments end up somewhere else, it is easier to see where they should go. Third, it would make dealing with the segments easier if someone has to deal with the segments outside Wubloader. Lastly, using the full datetime provides an extra layer of uniqueness for the segments in the case you have two segments that started at the same minute and second (but in different hours) and have the same content (or the unlikely event of a hash collision).

I don't see any real drawbacks to this.  The filenames are already long and not human friendly due to the hash. I believe this only requires a one line change to `downloader.make_path_prefix` as I don't think there is any code that tests against the time format of the segment.

This based on my own experiences that relying on directory structure to distinguish files can lead to mistakes.

**Comment by ekimekim** [comment]
My main concern is the loss of readability when viewing the segments as full paths, eg.
```
source/2018-12-16T10/59:55.751000-2.0-full-07YWX8WsKX5CMh90oxkl56323sfmIxB228by8CdMiIY.ts
source/2018-12-16T10/59:57.751000-2.0-full-VvlPKkgqWOVp2xapZE19Vzg0yZuZK9pk06mEG1gMmqY.ts
source/2018-12-16T10/59:59.751000-2.0-full-We5eJaW2RDKQthsue0YD6jto6AAv7nfJeLk19O8pRxg.ts
source/2018-12-16T11/00:01.751000-2.0-full-cVC1fhrWeWB46T14JMIfiSevPSfN1HSlPcdPLPF_V1g.ts
source/2018-12-16T11/00:03.751000-2.0-full-PixaZW3c2caapS07z550Za7VTODqb45Txfimzl_7wzk.ts
source/2018-12-16T11/00:05.751000-2.0-full-8adzvR6njpwieit-Ry9X0o-Uo4E3VdbRznoCOlTRWJo.ts
```
vs
```
source/2018-12-16T10/2018-12-16T10:59:55.751000-2.0-full-07YWX8WsKX5CMh90oxkl56323sfmIxB228by8CdMiIY.ts
source/2018-12-16T10/2018-12-16T10:59:57.751000-2.0-full-VvlPKkgqWOVp2xapZE19Vzg0yZuZK9pk06mEG1gMmqY.ts
source/2018-12-16T10/2018-12-16T10:59:59.751000-2.0-full-We5eJaW2RDKQthsue0YD6jto6AAv7nfJeLk19O8pRxg.ts
source/2018-12-16T11/2018-12-16T11:00:01.751000-2.0-full-cVC1fhrWeWB46T14JMIfiSevPSfN1HSlPcdPLPF_V1g.ts
source/2018-12-16T11/2018-12-16T11:00:03.751000-2.0-full-PixaZW3c2caapS07z550Za7VTODqb45Txfimzl_7wzk.ts
source/2018-12-16T11/2018-12-16T11:00:05.751000-2.0-full-8adzvR6njpwieit-Ry9X0o-Uo4E3VdbRznoCOlTRWJo.ts
```

My other concern is that now we have redundant information, and code needs to decide which one is used as canonical. If some code does it one way and some does it another, and then we somehow had a mismatch, we'd get very weird inconsistent behaviour. We could alternately always decode both and error out on mismatch, but that's adding cost to an operation that is already unacceptably slow on large ranges (asking for a 24h playlist can take upwards of 10sec to generate with the current code - I haven't profiled it yet but I wouldn't be surprised to know a signifigant chunk of that is date parsing).

As for the advantages: I don't really see a difference between a "file somehow ends up in a wrong directory" and "a file somehow ends up with the wrong filename". I guess conceptually I'd say the filepath is one string, and an error in one part of it (the directory) isn't that different to an error in another part (the filename).
Similarly, I'm 100% fine with two files in different hours having identical filenames (if somehow the start minutes/seconds, duration, and content were identical), because the full filepath is still unique.

Basically, the idea is that the entire directory structure isn't actually intended to be used as directories and files, it's just a quick and dirty way of creating an on-disk tree-structured key-value store.

The possibility of needing to deal with segments outside wubloader is a good point, but I don't know if it's worth the additional ambiguity in parsing. And if we were to go down that route, I'd argue that by the same logic we should also be including stream and variant info in the filename, so the entire directory structure is replicated to the filename.

**Comment by chrusher** [comment]
Fair enough. I'll close this.
